src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -130,14 +131,42 @@ private CallbackResponse validateApplicantExperts(CallbackParams callbackParams)
     }
 
     private CallbackResponse setApplicant1ProceedFlag(CallbackParams callbackParams) {
-        CaseData caseData = callbackParams.getCaseData();
+        CaseData updatedCaseData = setApplicant1ProceedFlagToYes(callbackParams.getCaseData());
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.toMap(objectMapper))
+            .build();
+    }
+
+    private CaseData setApplicant1ProceedFlagToYes(CaseData caseData) {
         var updatedCaseData = caseData.toBuilder();
+
         if (TWO_V_ONE.equals(getMultiPartyScenario(caseData))
             && YES.equals(caseData.getApplicant1ProceedWithClaimSpec2v1())) {
             updatedCaseData.applicant1ProceedWithClaim(YES);
         }
+        return updatedCaseData.build();
+    }
+
+    private CaseData setApplicantDefenceResponseDocFlag(CaseData caseData) {
+        var updatedCaseData = caseData.toBuilder();
+
+        if (NO.equals(caseData.getApplicant1AcceptAdmitAmountPaidSpec())
+            || (caseData.getRespondent1ClaimResponseTypeForSpec().equals(RespondentResponseTypeSpec.FULL_DEFENCE)
```

Comment:
i would still put this in a boolean method that is more descriptive of what the condition is you are testing. sorry to be picky

Reply:
done